functions/src/definitions/common/repostUtils.ts

```diff
@@ -0,0 +1,133 @@
+/*
+Stores a collection of functions for reposting messages to a admin channel
+*/
+import { AppEnv } from "../../appEnv"
+import {
+  sendTelegramImageMessage,
+  sendTelegramTextMessage,
+  updateTelegramTextMessage,
+  updateTelegramImageMessage,
+} from "./sendTelegramMessage"
+import * as admin from "firebase-admin"
+import { getSignedUrl } from "./mediaUtils"
+
+const db = admin.firestore()
+const repostChannelId = process.env[AppEnv.TELEGRAM_REPOST_CHANNEL_ID] || ""
+
+// Stores the replyId and instanceText whenever we repost a message
+export async function createMessageId(
+  messageId: string,
+  replyId: string,
+  instanceText: string
+) {
+  try {
+    const repostIdRef = db.collection("repostIds").doc(messageId)
```

Comment:
Avoid creating new collection for this.

Since we're tying repost to messages, put the repostId as a field in the message object itself. Then the text/caption is also easily accessible cos it's already there.